src/segmantic/seg/monai_unet.py

```diff
@@ -803,3 +809,149 @@ def cross_validate(
                         )
             else:
                 continue
+
+
+def ensemble_evaluate(post_transforms, models, device, test_loader):
+    evaluator = EnsembleEvaluator(
+        device=device,
+        val_data_loader=test_loader,
+        pred_keys=["pred0", "pred1", "pred2"],
+        networks=models,
+        inferer=SlidingWindowInferer(
+            roi_size=(96, 96, 96), sw_batch_size=4, overlap=0.5
+        ),
+        postprocessing=post_transforms,
+        # key_val_metric={
+        #    "test_mean_dice": MeanDice(
+        #        include_background=False,
+        #        output_transform=from_engine(["pred", "label"]),
+        #    )
+        # },
+    )
+    evaluator.run()
+
+
+def ensemble_creator(
+    model_files: List[Path],
+    test_images: List[Path],
+    test_labels: Optional[List[Path]] = None,
+    output_dir: Path = None,
+    tissue_dict: Dict[str, int] = None,
+    spacing: Sequence[float] = [],
+    combination_mode: str = "SelectBest",
+    gpu_ids: List[int] = [],
+):
+    device = make_device(gpu_ids)
+
+    models = [Net.load_from_checkpoint(str(ckpt_path)) for ckpt_path in model_files]
+
+    print(tissue_dict)
+
+    if test_labels:
+        assert len(test_images) == len(test_labels)
+        test_files = [
+            {"image": i, "label": l} for i, l in zip(test_images, test_labels)
+        ]
+    else:
+        test_files = [{"image": i} for i in test_images]
+
+    pre_transforms = Compose(
+        [
+            models[0].default_preprocessing(
+                keys=["image", "label"] if test_labels else ["image"],
+                spacing=spacing,
+            ),
+            # AsDiscreted(keys="label", to_onehot=models[0].num_classes),
+        ]
+    )
+
+    # data loader
+    test_loader = DataLoader(
+        Dataset(
+            data=test_files,
+            transform=pre_transforms,
+        ),
+        batch_size=1,
+        num_workers=0,
+    )
+
+    # save predicted labels
+    save_transforms: List[MapTransform] = []
+    if output_dir:
+        os.makedirs(output_dir, exist_ok=True)
+
+        save_transforms = [
+            SaveImaged(
+                keys="pred",
+                output_dir=output_dir,
+                output_postfix="seg",
+                resample=False,
+                separate_folder=False,
+                print_log=False,
+                writer="ITKWriter",
+            )
+        ]
+    if combination_mode == "Mean":
```

Comment:
I suggest you use enums (they work nicely with typer for command lines), e.g. 

```python
from enum import Enum

class EnsembleCombination(str, Enum):
    mean = "mean"
    vote = "vote"
    best = "best"


def ensemble_creator(
    ...,
    combination_mode: EnsembleCombination = EnsembleCombination.best.value
```

Reply:
Good idea thank you~~